cli/command/image/build.go

```diff
@@ -146,7 +146,16 @@ func NewBuildCommand(dockerCli command.Cli) *cobra.Command {
 	flags.StringVar(&options.imageIDFile, "iidfile", "", "Write the image ID to the file")
 
 	command.AddTrustVerificationFlags(flags, &options.untrusted, dockerCli.ContentTrustEnabled())
-	command.AddPlatformFlag(flags, &options.platform)
+
+	flags.StringVar(&options.platform, "platform", os.Getenv("DOCKER_DEFAULT_PLATFORM"), "Set platform if server is multi-platform capable")
+	// Platform is not experimental when BuildKit is used
+	buildkitEnabled, err := command.BuildKitEnabled(dockerCli.ServerInfo())
```

Comment:
I think this is buggy:
When we are creating commands and flags, we have not yet run PersistentPreRun, and so I think the CLI is not yet correctly initialized. I think `docker --context=some-context build --platform ...`  can then have a very wierd behavior (where flag support is evaluated for a daemon that is not the one used after flags are actually parsed)
WDYT @thaJeztah @vdemeester @silvin-lubecki  ?